txpool/queue.go

```diff
@@ -0,0 +1,107 @@
+package txpool
+
+import (
+	"container/heap"
+
+	"github.com/0xPolygon/polygon-sdk/types"
+)
+
+/* Queue implementations required by the heap interface */
+
+type transactions []*types.Transaction
```

Comment:
I don't think you necessarily need to create a type with no specific characteristics/behavior just because it's possible and I would argue that the benefit of improved readability is insignificant because I don't know that `transactions` is simply a slice until go to the type definition and see nothing specific about it. Not a big deal in general, it's up to you to decide the destiny of this type :)

Reply:
If the cost of not having sore eyes is one type definition lookup, I'm more that willing to pay for it.

Reply:
I'm with @brkomir on this one. We should generally adopt that approach in our code-style: no type definitions that have no special methods on them.

Providing functional abstractions is a good thing, but this is not one of those cases, as we are simply hiding from plainsight the important information about a type being a slice of pointers to transactions.